### PR TITLE
Add mactex-no-ghostscript 20170524

### PR DESCRIPTION
**Pull request by commitay**
After making all changes to the cask:

- [ ] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **updating a cask**:

- [ ] `sha256` changed but `version` stayed the same ([what is this?][version-checksum]).  
      I’m providing public confirmation below.

Additionally, if **adding a new cask**:

- [ ] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not already refused in [closed issues].
- [ ] Checked the cask is submitted to [the correct repo].

https://github.com/caskroom/homebrew-versions/pull/4352

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/mactex-no-ghostscript.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                   'choiceIdentifier' => 'choice2',
+                   'choiceAttribute'  => 'selected',
+                   'attributeSetting' => 1,
+                 },
+                 {
+                   'choiceIdentifier' => 'choice3',
+                   'choiceAttribute'  => 'selected',
+                   'attributeSetting' => 0,
+                 },
+               ]
```
Comment (reitermarkus) [inline Casks/mactex-no-ghostscript.rb]: These should have comments, since `choice[123]` isn't obvious.

File: Casks/mactex-no-ghostscript.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                '~/Library/TeXShop',
+              ],
+      rmdir:  [
+                '/usr/local/texlive',
+                '~/Library/texlive',
+              ]
+
+  caveats do
+    <<-EOS.undent
+      This cask installs Homebrew ghostscript instead of the mactex packaged ghostscript.
```
Comment (reitermarkus) [inline Casks/mactex-no-ghostscript.rb]: Uppercase Cask and MacTex. And maybe reword it: “This Cask installs `ghostscript` from Homebrew instead of the one packaged with MacTex.”

File: Casks/mactex-no-ghostscript.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                '~/Library/texlive/2017',
+                '~/Library/TeXShop',
+              ],
+      rmdir:  [
+                '/usr/local/texlive',
+                '~/Library/texlive',
+              ]
+
+  caveats <<-EOS.undent
+    This Cask installs ghostscript from Homebrew instead of the one packaged with MacTex.
```
Comment (miccal) [inline Casks/mactex-no-ghostscript.rb]: But it does not install `ghostscript` automatically?

Comment (commitay) [reply Casks/mactex-no-ghostscript.rb]: I forgot to add the depends on formula.

Comment (commitay) [reply Casks/mactex-no-ghostscript.rb]: Fixed.

**Review by miccal — APPROVED** [review]

**Review by reitermarkus — APPROVED** [review]

File: Casks/mactex-no-ghostscript.rb (hunk 3 above)
Comment (jlaw) [reply Casks/mactex-no-ghostscript.rb]: Please change the capitalization to MacTeX.

**Comment by Suavatore** [comment]
Hi, so is mactex-no-ghostscript complete and safe to install now? Just want to make sure.

**Comment by eddieschoute** [comment]
Would it not be possible to have a flag (e.g. `--no-ghostscript`) with the mactex cask instead? This feels like duplication of the package.

**Comment by commitay** [comment]
> Would it not be possible to have a flag (e.g. `--no-ghostscript`) with the mactex cask instead?

No, we don't support flags like this. Very few Casks would benefit from such a feature.

> This feels like duplication of the package.

Yes, it is the only way we can provide this Cask.
